extensions/wikia/PolldaddyTag/PolldaddyTagController.class.php

```diff
@@ -0,0 +1,50 @@
+<?php
+class PolldaddyTagController extends WikiaController {
+
+	const PARSER_TAG_NAME = 'polldaddy';
+	const NAME = 'Polldaddy';
+
+	const DATA_WIKI_WIDGET_ATTRIBUTE = [
+		'data-wikia-widget' => self::PARSER_TAG_NAME,
+	];
+
+	private $helper;
+	private $validator;
+
+	public function __construct() {
+		parent::__construct();
+
+		$this->helper = new WikiaIFrameTagBuilderHelper();
+		$this->validator = new PolldaddyTagValidator();
+	}
+
+	public static function onParserFirstCallInit( Parser $parser ) {
+		$parser->setHook( self::PARSER_TAG_NAME, [ new static(), 'renderTag' ] );
+		return true;
+	}
+
+	public function renderTag( $input, array $args, Parser $parser, PPFrame $frame ) {
+		$isValid = $this->validator->validateAttributes( $args );
+
+		if ( !$isValid ) {
+			return '<strong class="error">' . wfMessage( 'polldaddy-tag-could-not-render' )->parse() . '</strong>';
+		}
+
+		return $this->helper->isMobileSkin() ?
+			Html::rawElement(
+				'a',
+				array_merge( self::DATA_WIKI_WIDGET_ATTRIBUTE, [ 'data-id' => $args['id'] ] )
+			) :
+			Html::rawElement(
+				'span',
+				self::DATA_WIKI_WIDGET_ATTRIBUTE,
```

Comment:
We need to have a conversation (including @justnpT) about this `data-wikia-widget` attribute and where we're adding it and do some code cleanup. Why? It differs along the widgets and we like consistency — especially when writing selenium-tests. :coffee: 
* Simple inline framed widgets have the `data-` attribute added to the `<iframe>` — simple indeed.
* VK has it added to the `<div>` inside the `<span>` wrapper; This wrapper has a `widget-vk` class.
* Twitter has it added the `<a>` inside the `<span>` wrapper; This wrapper has a `widget-twitter` class, but the aforementioned `<a>` tag is swapped by the Twitter script for an `<iframe>`. We don't control this `<iframe>`.
* Here you add it to the wrapper… and IMHO it's the best way. :smiley: 

Also I'm not sure about adding the wrapper to the VK widget, as we control the `<div>` which the VK script looks for and adds the `<iframe>` into it — we naturally have a wrapper.